processing/src/main/java/io/druid/query/filter/BoundDimFilter.java

```diff
@@ -0,0 +1,182 @@
+package io.druid.query.filter;
+
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+import com.metamx.common.StringUtils;
+
+import java.nio.ByteBuffer;
+
+public class BoundDimFilter implements DimFilter
+{
+  private final String dimension;
+  private final String upper;
+  private final String lower;
+  private final boolean lowerStrict;
+  private final boolean upperStrict;
+  private final boolean alphaNumeric;
+
+  @JsonCreator
+  public BoundDimFilter(
+      @JsonProperty("dimension") String dimension,
+      @JsonProperty("lower") String lower,
+      @JsonProperty("upper") String upper,
+      @JsonProperty("lowerStrict") Boolean lowerStrict,
+      @JsonProperty("upperStrict") Boolean upperStrict,
+      @JsonProperty("alphaNumeric") Boolean alphaNumeric
+  )
+  {
+    this.dimension = Preconditions.checkNotNull(dimension, "dimension can not be null");
+    Preconditions.checkState((lower != null) || (upper != null), "lower and upper can not be null at the same time");
+    this.upper = upper;
+    this.lower = lower;
+    this.lowerStrict = (lowerStrict == null) ? false : lowerStrict;
+    this.upperStrict = (upperStrict == null) ? false : upperStrict;
+    this.alphaNumeric = (alphaNumeric == null) ? false : alphaNumeric;
+  }
+
+  @JsonProperty
+  public String getDimension()
+  {
+    return dimension;
+  }
+
+  @JsonProperty
+  public String getUpper()
+  {
+    return upper;
+  }
+
+  @JsonProperty
+  public String getLower()
+  {
+    return lower;
+  }
+
+  @JsonProperty
+  public boolean isLowerStrict()
+  {
+    return lowerStrict;
+  }
+
+  @JsonProperty
+  public boolean isUpperStrict()
+  {
+    return upperStrict;
+  }
+
+  @JsonProperty
+  public boolean isAlphaNumeric()
+  {
+    return alphaNumeric;
+  }
+
+  @Override
+  public byte[] getCacheKey()
+  {
+    byte[] dimensionBytes = StringUtils.toUtf8(this.getDimension());
+    byte[] lowerBytes = StringUtils.toUtf8("lower:" + this.getLower());
+    byte[] upperBytes = StringUtils.toUtf8("upper:" + this.getUpper());
```

Comment:
not sure why is encoding "lower/upper" necessary, it will only increase size of the key. you could use something like..
```
byte[] lowerBytes = (lower == null) ? EMPTY_BYTE_ARR : StringUtils.toUtf8(lower);
```
also, you wouldn't need any of the if statements later.